core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -249,4 +268,9 @@ final public boolean hasTop() {
   public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public void close() throws Exception {
+    sources.forEach(s -> s.closeSafely());
+  }
```

Comment:
Probably, yeah. If any one throws an exception, you can remember that, and either aggregate them into a single exception to throw at the end, or just propagate the last one seen.